pkg/core/mpt/billet.go

```diff
@@ -242,12 +242,32 @@ func (b *Billet) traverse(curr Node, path, from []byte, process func(pathToNode
 		var (
 			startIndex byte
 			endIndex   byte = childrenCount
+			inc             = func(i byte) byte {
+				i++
+				return i
+			}
+			cmp = func(i byte) bool {
+				return i < endIndex
+			}
 		)
+		if backwards {
+			startIndex, endIndex = endIndex, startIndex
+			inc = func(i byte) byte {
+				i--
+				return i
+			}
+			cmp = func(i byte) bool {
+				return i >= startIndex
+			}
+		}
 		if len(from) != 0 {
 			endIndex = lastChild
+			if backwards {
+				endIndex = 0
+			}
 			startIndex, from = splitPath(from)
```

Comment:
`splitPath` can set `startIndex` to 0 and `i` is unsigned. Right now it is not obvious why backwards iteration exit condition `i >= startIndex` should ever be false in this case. Can this in fact happen? If not, could you add a comment about it?

Reply:
You're right, fixed.